File: packages/python-sdk/src/blindfold/regex/detectors/uk/utr.py
"""UK Unique Taxpayer Reference (UTR) detector"""

import re

from ...base import RegexDetector
from ...entities import EntityType
from ...registry import register_region
from ...validators import uk_utr_checksum


@register_region("uk")
class UtrDetector(RegexDetector):
    entity_type = EntityType.UK_UTR
    score = 0.85
    context_keywords = [
        "utr", "unique taxpayer", "tax reference", "self assessment", "hmrc",
    ]
    context_required = True
    context_window = 50

    # 10-digit UTR number
    pattern = re.compile(
        r"\b\d{10}\b"
    )
    validator = staticmethod(uk_utr_checksum)

File: packages/js-sdk/src/regex/detectors/uk/utr.ts
// UK Unique Taxpayer Reference (UTR) detector

import { RegexDetector } from '../../base'
import { EntityType } from '../../entities'
import { registerRegion } from '../../registry'
import { ukUtrChecksum } from '../../validators'

class UkUtrDetector extends RegexDetector {
  entityType = EntityType.UK_UTR
  score = 0.85
  contextKeywords = ['utr', 'unique taxpayer', 'tax reference', 'self assessment', 'hmrc']
  contextRequired = true

  pattern = /\b\d{10}\b/g
  validator = ukUtrChecksum
}

registerRegion('uk', UkUtrDetector)

export { UkUtrDetector }
